source/slang/slang-check-stmt.cpp

```diff
@@ -327,6 +329,260 @@ namespace Slang
         stmt->expression = CheckExpr(stmt->expression);
     }
 
+    void SemanticsStmtVisitor::tryInferLoopMaxIterations(ForStmt* stmt)
+    {
+        DeclRef<Decl> initialVar = {};
+        DeclRef<Decl> predicateVar = {};
+        Expr* initialVal = nullptr;
+        if (auto varStmt = as<DeclStmt>(stmt->initialStatement))
```

Comment:
Might benefit with some comments saying what this does and doesn't handle. 
I'm assuming doing (say) ```for (int i = 1, j = 2;... )``` is something it just doesn't handle, but I'm not 100% sure what that maps down to in AST.

Reply:
Added a comment at the top of the function.